### PR TITLE
Shared physical file provider with base directory as root

### PR DESCRIPTION
**Pull request by mapogolions**

### CODE REVIEW

File: src/libraries/Microsoft.Extensions.Configuration.FileExtensions/src/FileConfigurationExtensions.cs [hunk 1]
```diff
@@ -47,8 +47,9 @@ public static IFileProvider GetFileProvider(this IConfigurationBuilder builder)
             {
                 return provider as IFileProvider;
             }
-
-            return new PhysicalFileProvider(AppContext.BaseDirectory ?? string.Empty);
+            var fileProvider = new PhysicalFileProvider(AppContext.BaseDirectory ?? string.Empty);
+            builder.Properties[FileProviderKey] = fileProvider;
```
Comment (eerhardt) [inline src/libraries/Microsoft.Extensions.Configuration.FileExtensions/src/FileConfigurationExtensions.cs:51]: I'm not certain this is a valid change. Thinking about https://github.com/dotnet/runtime/issues/63540, the `PhysicalFileProvider` is `IDisposable`. Sticking it in the `builder.Properties` means people may be able to get at it after it has been disposed.

Can you add a reasoning/scenario (maybe in an issue?) for why this change is necessary?

Comment (mapogolions) [reply src/libraries/Microsoft.Extensions.Configuration.FileExtensions/src/FileConfigurationExtensions.cs:51]: As I understand from code, the `Properties` of the `IConfigurationBuilder` instance is used like a temporary bag to store some shared data that can be used during construction the `IConfiguration` . Sticking the `PhysicalFileProvider` in the `Properties` of the `ConfigurationBuilder` also happens  in case when we use another [extension method the `SetBasePath`](https://github.com/mapogolions/runtime/blob/ce2165d8084cca98b95f5d8ff9386759bfd8c722/src/libraries/Microsoft.Extensions.Configuration.FileExtensions/src/FileConfigurationExtensions.cs#L72). Could you explain to me what the difference between these cases?

Comment (eerhardt) [reply src/libraries/Microsoft.Extensions.Configuration.FileExtensions/src/FileConfigurationExtensions.cs:51]: > Could you please explain to me what the difference between these cases?

The difference is that calling `SetBasePath` is an explicit action to stick the file provider into the `Properties` collection. You can use a single `IConfigurationBuilder` to build multiple configurations.

If someone disposed the `PhysicalFileProvider` while it was being held on to the `ICollectionBuilder`, you can get an `ObjectDisposedException` if you tried using the `ICollectionBuilder` again. It should be an explicit action to cache the file provider into the `ICollectionBuilder`.
